presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -160,9 +149,18 @@ private boolean hasPreviousSpillCompletedSuccessfully()
         }
     }
 
-    private boolean shouldSpill(long memorySize)
+    @Override
+    public ListenableFuture<?> startMemoryRevoke()
     {
-        return (memorySizeBeforeSpill > 0 && memorySize > memorySizeBeforeSpill);
```

Comment:
Should the `memoryLimitBeforeSpill` variable in `com.facebook.presto.sql.planner.LocalExecutionPlanner.Visitor#visitAggregation` be removed too? I understand now there is no hard memory limit before spill.